scripts/zones/Windurst_Walls/npcs/Arbitrix.lua

```diff
@@ -7,47 +7,210 @@
+                    player:updateEvent(itemid, 5, 0)
+                else
+                    player:updateEvent(1, 5, 1) -- not enough points
+                end
+            end,
+            [42] = function()
+                if player:getFreeSlotsCount() == 0 then
+                    player:updateEvent(holdingItem, 0, 0, 1) -- inventory full, exit event
+                    player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED + 2) -- generic "Cannot obtain the item."
+                end
+            end,
+            [45] = function() -- Dial 5 Finish
+                if dailyTallyPoints >= 10 then
+                    if npcUtil.giveItem(player, holdingItem) then
+                        player:setCharVar("gobbieBoxHoldingItem", 0)
+                    end
+                end
+                player:updateEvent(specialDialUsed, adoulinDialUsed, pictlogicaDialUsed, wantedDialUsed, 0, 0, hideOptionFlags, player:getCurrency("daily_tally"))
+            end,
+            [49] = function() -- Special Dial
+                if specialDialUsed == 1 then
+                    player:updateEvent(1, 6, 2) -- already used this dial
+                elseif dailyTallyPoints >= 50 then
+                    itemid = player:selectDailyItem(6)
+                    player:setCharVar("gobbieBoxHoldingItem", itemid)
+                    player:setCurrency("daily_tally", dailyTallyPoints - 50)
+                    player:setMaskBit(gobbieBoxUsed, "gobbieBoxUsed", 0, true)
+                    player:updateEvent(itemid, 6, 0)
+                else
+                    player:updateEvent(1, 6, 1) -- not enough points
+                end
+            end,
+            [50] = function()
+                if player:getFreeSlotsCount() == 0 then
+                    player:updateEvent(holdingItem, 0, 0, 1) -- inventory full, exit event
+                    player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED + 2) -- generic "Cannot obtain the item."
+                end
+            end,
+            [53] = function() -- Special Dial Finish
+                if dailyTallyPoints >= 50 then
+                    if npcUtil.giveItem(player, holdingItem) then
+                        player:setCharVar("gobbieBoxHoldingItem", 0)
+                    end
+                end
+                player:updateEvent(specialDialUsed, adoulinDialUsed, pictlogicaDialUsed, wantedDialUsed, 0, 0, hideOptionFlags, player:getCurrency("daily_tally"))
+            end,
+        }
     end
 end
 
 function onEventFinish(player, csid, option)
     if csid == 519 then
         player:setCurrency("daily_tally", 50)
+    elseif csid == 521 then
+        if player:getFreeSlotsCount() == 0 then
+            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED + 2) -- generic "Cannot obtain the item."
+        elseif npcUtil.giveItem(player, player:getCharVar("gobbieBoxHoldingItem")) then
+            player:setCharVar("gobbieBoxHoldingItem", 0)
```

Comment:
`npcUtil.giveItem` will handle the slot count and displaying unable to obtain for you:
```lua
if not npcUtil.giveItem(player, player:getCharVar("gobbieBoxHoldingItem") then
    player:setCharVar("gobbieBoxHoldingItem", 0)
end
```

Reply:
npcUtil.giveItem actually gives a different message than this. "Unable to obtain <item>." vs "Unable to obtain the item."

Reply:
Oh right, the mystery box doesn't say what the item is, does it? Nevermind then!